## Data Understanding
### Data Loading

In [1]:
# import library
import pandas as pd
import numpy as np
import scipy
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
# Convert into dataframe
df = pd.read_excel("../data/raw/Flipkart-Laptops.xlsx")

# Top five observations
df.head()

,Product Name,ProductID,Product image,Actual price,Discount price,Stars,Rating,Reviews,Description,Link
0,MSI Cyborg 15 Intel Core i5 12th Gen 12450H - ...,COMGZW35W3DSJADN,NaN,89990,54990,3.9,7 Ratings,1 Reviews,Intel Core i5 Processor (12th Gen)16 GB DDR5 R...,https://www.flipkart.com/msi-cyborg-15-intel-c...
1,MSI Thin 15 Intel Core i7 12th Gen 12650H - (8...,COMGZW37ZX66DBHF,NaN,83990,67990,NIL,NIL,NIL,Intel Core i7 Processor (12th Gen)8 GB DDR4 RA...,https://www.flipkart.com/msi-thin-15-intel-cor...
2,DELL Inspiron 3520 Intel Core i3 12th Gen 1215...,COMGJ75HJGFDJ6JN,NaN,49240,35660,4.2,"1,805 Ratings",143 Reviews,Intel Core i3 Processor (12th Gen)8 GB DDR4 RA...,https://www.flipkart.com/dell-inspiron-3520-in...
3,Acer One (2024) Intel Core i3 11th Gen 1115G4 ...,COMGPF5CQ7VDWDT4,NaN,43999,26990,4.2,"6,977 Ratings",596 Reviews,Intel Core i3 Processor (11th Gen)8 GB DDR4 RA...,https://www.flipkart.com/acer-one-2024-intel-c...
4,Lenovo V15 AMD Ryzen 3 Quad Core 7320U - (8 GB...,COMGPYKZAWY8UX6C,NaN,59400,27989,4.2,"1,263 Ratings",113 Reviews,AMD Ryzen 3 Quad Core Processor8 GB LPDDR5 RAM...,https://www.flipkart.com/lenovo-v15-amd-ryzen-...


In [5]:
# Shape of the dataset
print(df.shape)

(960, 10)


In [6]:
# Basic information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 960 entries, 0 to 959
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Product Name    960 non-null    str    
 1   ProductID       960 non-null    str    
 2   Product image   0 non-null      float64
 3   Actual price    960 non-null    object 
 4   Discount price  960 non-null    object 
 5   Stars           960 non-null    object 
 6   Rating          960 non-null    str    
 7   Reviews         960 non-null    str    
 8   Description     960 non-null    str    
 9   Link            960 non-null    str    
dtypes: float64(1), object(3), str(6)
memory usage: 75.1+ KB


In [7]:
## Check total no. of null values in each column
df.isnull().sum()

Product Name        0
ProductID           0
Product image     960
Actual price        0
Discount price      0
Stars               0
Rating              0
Reviews             0
Description         0
Link                0
dtype: int64

In [8]:
## Total no. of duplicate values
df.duplicated().sum()

np.int64(0)

**Comments:**

we can see that there are 960 observations and 10 columns. In Product image column, there are no values available, i.e., all are nan values. we can also observe that, except Product image column, all columns are object type.